daemon/containerd/image_exporter.go

```diff
@@ -310,8 +318,9 @@ func (i *ImageService) LoadImage(ctx context.Context, inTar io.ReadCloser, platf
 				return nil
 			}
 
-			// Only unpack the image if it matches the host platform
-			if !i.hostPlatformMatcher().Match(imgPlat) {
+			// Only unpack the image if it matches the requested platform or
+			// the host platform (if no specific platform was requested).
+			if !pm.Match(imgPlat) {
```

Comment:
This is to prevent unpacking to fail;

```
docker image load -i image-multiple.tar --platform=linux/s390x
Loaded image: alpine:latest
Error unpacking image alpine:latest: NotFound: content digest sha256:cf04c63912e16506c4413937c7f4579018e4bb25c272d989789cfba77b12f951: not found
```